### DISCUSSION TITLE
PR merge policy

### DISCUSSION THREAD

**Discussion by Consolatis** [labwc/labwc#834] [Team Posts]
Merging own PRs is fine if at least one of the following matches:
- PR only containing minor fixups that are not expected to break anything
- PR only containing a minimal obvious crash fix (which includes triggered asserts)
- At least one other member of the labwc orga reviewed and put its LGTM behind it
- The change is about documentation / CI / comments or similar that doesn't change the code itself
- New features (regardless of how invasive they are) should \*not\* be merged if there is no LGTM by another member of the labwc orga

Merging PRs of others:
- Look through the code and if required do a quick test run nested
- Try to make sure the code syntax matches the rest of the project
- If something is required to be tested depends on the judgement of the reviewer
- If the PR is from another member of the orga either leave your LGTM (+ potentially test status) in there or merge it directly if the PR is not in Draft mode.

`Rebase/Merge` vs. `Merge` vs `Squash`:
- Per default always use `Rebase/Merge` to prevent useless merge commits in the history
- If a PR is discussed a lot and that discussion provides further reasoning for the PR or explains why something is done in a specific way use `Merge` instead so we have a link from the commit history to a PR number.

Review process:
- Having different opinions about code style is fine, if both sides (committer + reviewer)  can't come to a conclusion it is absolutely fine (and expected) to elevate the issue to the project lead @johanmalm.
- PRs that have open issues should be moved to Draft mode to prevent other members to accidentally merge them while they are still being worked on.
- Remember that the review process is not about showing off who is the better coder but is about to keep / increase the quality and especially maintainability of the code for the future. Think about what would have made it easier for you to understand the code when you first looked at it.

Update of the merge policy:
- If something should be added / removed / reworded / adjusted in this policy feel free to add a comment.

**Comment by johanmalm** [comment]
Looks really good.

**Comment by Consolatis** [comment]
Any suggestion for enhancements / modifications?
Otherwise I'd create a copy of the post and pin it.

Also: @jlindgren90 welcome to the dev team :)

**Comment by jlindgren90** [comment]
LGTM, and thanks!

**Comment by johanmalm** [comment]
Yes go for it.
